Implement New Features and Updates in feat-omidupdate Branch

OmidZamani opened the pull request:
- Added new initialization script for Gaia with dynamic environment setup
- Updated check_version.sh script

OmidZamani committed:
Added new initialization script for Gaia with dynamic environment setup

OmidZamani committed:
Updated check_version.sh script

OmidZamani committed:
Updated Dockerfile and entrypoint script in osmosis directory

OmidZamani committed:
Update Dockerfile in osmosis directory

OmidZamani committed:
Refactor Dockerfile, entrypoint.sh, and check_version.sh for Variable Consistency

- Updated Dockerfile to align with ARG and ENV variable changes for better build process.
- Modified entrypoint.sh to synchronize environment setup with Dockerfile variables and ensure proper initialization of Gaia.
- Revised check_version.sh to enhance version checking logic and ensure compatibility with updated Dockerfile and entrypoint.sh configurations.
- Ensured all scripts are in sync with the latest variable definitions and dependencies.

mehrdadpfg commented on cosmos/Dockerfile:
```diff
@@ -1,9 +1,8 @@
 # Stage 1: Build stage
 ARG GO_VERSION
 FROM golang:$GO_VERSION-alpine AS go-builder
-ARG arch=x86_64
```
Should not remove arch

mehrdadpfg commented on cosmos/entrypoint.sh:
```diff
@@ -1,35 +1,31 @@
-envsubst < $HOME/config-sample/client.toml > $CONFIG_PATH/client.toml
-envsubst < $HOME/config-sample/config.toml > $CONFIG_PATH/config.toml
+# Copy config files
+cp $HOME/config-sample/* $CONFIG_PATH/
```
No need for this
Config folder should be removed as well in the repo

OmidZamani committed:
Update Docker configuration and entrypoints for Cosmos ecosystem

- Modified Dockerfiles for Cosmos and Stride to enhance containerization.
- Updated entrypoint scripts for improved startup procedures.
- Removed outdated configuration files (app.toml, client.toml, config.toml) across multiple directories (Cosmos, Osmosis, Stride, Terra) to align with the latest project standards.

OmidZamani committed:
Update Dockerfiles for Cosmos, Osmosis, Stride, and Terra Networks

- Added ARG for BIN_VERSION and WASM_VERSION to Dockerfiles
- Ensured compatibility of BIN_VERSION and WASM_VERSION across all networks
- Adjusted Dockerfile structure to align with new variable requirements
- Removed unnecessary 'arch' variable from Dockerfiles
- Streamlined build process for each network's specific needs

OmidZamani committed:
delete config folder

OmidZamani committed:
Refactor Dockerfiles and Entrypoints Across Networks

- Updated Dockerfiles for Cosmos, Osmosis, Stride, and Terra to ensure consistency and correct versioning.
- Adjusted entrypoint scripts to align with Dockerfile changes and network-specific configurations.
- Added necessary environment variables and parameters for each network.
- Removed unnecessary commands and streamlined the setup process.
- Ensured compatibility of versions and dependencies across all networks.

OmidZamani committed:
edit entrypoint.sh

OmidZamani committed:
Add Binary Initialization to entrypoint.sh for Osmosis

This commit includes modifications in the entrypoint.sh script to add the initial binary configuration step (Binary Initialization) for the Osmosis node. These changes involve executing the `osmosisd init` command with the appropriate chain ID for the environment (mainnet or testnet), read from the `info` file. These adjustments are made to align with project requirements and management directives.

mehrdadpfg commented on cosmos/Dockerfile:
```diff
@@ -4,6 +4,7 @@ FROM golang:$GO_VERSION-alpine AS go-builder
 ARG arch=x86_64
 
 ARG BIN_VERSION
+ARG WASM_VERSION
```
cosmos do not need wasm why here??!

mehrdadpfg commented on osmosis/entrypoint.sh:
```diff
@@ -1,39 +1,16 @@
-    tar -xjf genesis.tar.bz2 && rm genesis.tar.bz2
-  fi
+if [ "$ENVIRONMENT" == "mainnet" ]; then
+    osmosisd init --chain-id $MAINNET_CHAIN_ID
```
what happend to downloading genesis and addrbook and snapshot?????

mehrdadpfg commented on stride/Dockerfile:
```diff
@@ -20,29 +23,23 @@ RUN git clone --depth 1 https://github.com/microsoft/mimalloc; \
     chown stride:stride -R /stride
 
-
 COPY ./config /stride/config-sample
```
remove line that copies config file there is no config file

mehrdadpfg commented on terra/Dockerfile:
```diff
@@ -1,12 +1,10 @@
 
-ENV TERRA_VERSION=v2.5.2
-ENV CHAIN_ID=phoenix-1
-ENV LIBWASMVM_VERSION=v1.4.1
```
removed this env where is the replacement? where is wasm version??

mehrdadpfg commented on terra/Dockerfile:
```diff
@@ -24,39 +22,30 @@ RUN git clone --depth 1 https://github.com/microsoft/mimalloc; \
 
-# Install libraries and verify checksums
-ADD https://github.com/CosmWasm/wasmvm/releases/download/${LIBWASMVM_VERSION}/libwasmvm_muslc.aarch64.a libwasmvm_muslc.aarch64.a
-ADD https://github.com/CosmWasm/wasmvm/releases/download/${LIBWASMVM_VERSION}/libwasmvm_muslc.x86_64.a libwasmvm_muslc.x86_64.a
```
why these are removed?

OmidZamani committed:
Update Osmosis entrypoint script for mainnet and testnet setup

- Added logic to fetch chain IDs from the 'info' file for both mainnet and testnet.
- Implemented download and setup of necessary files (snapshot, addrbook, genesis) for Osmosis mainnet.
- Added download and setup steps for Osmosis testnet, including seeds and persistent peers configuration.
- Ensured proper initialization of Osmosis node with 'osmosisd init' command based on the environment.
- Initiated Osmosis daemon with 'osmosisd start'.

OmidZamani committed:
Remove config file copying from Stride Dockerfile

This commit removes the line in the Stride Dockerfile that was responsible for copying configuration files from the config-sample directory to the container. This change aligns with the project requirements to eliminate unnecessary configuration file copying, simplifying the Dockerfile and ensuring configurations are managed externally.

OmidZamani committed:
Remove config directory creation from Dockerfiles

- Removed lines related to config directory creation in Dockerfiles of Cosmos, Osmosis, Terra, and Stride networks.
- Ensured other necessary directories and user settings remain intact.
- Adjusted to align with project requirements and manager's feedback.

OmidZamani committed:
remove line 37 - mkdir -p /gaia/.gaia/data && \
